content/2019-01-20-pythran-0.9.1.rst

```diff
@@ -0,0 +1,226 @@
+
+.. code-block:: python
+
+    #pythran export broadcast_and_slice(float[:,:,:], float[:])
+    def broadcast_and_slice(x, y):
+        return (x + y) [1:]
+
+It's a tricky one because as a result of broadcasting (`x` and `y` don't have the same number of dimension) Numpy creates a temporarily large array, and right after it slices it. Pythran can now evaluates this expression lazily and avoid the creation of the intermediate (large) array.
+
+Support ``numpy.stack``, ``numpy.rollaxis``, ``numpy.broadcast_to`` and ``numpy.ndarray.dtype.type``
+----------------------------------------------------------------------------------------------------
+
+Well, the title says it all. The Numpy API is huge but we're moving forward.
+
+Better support of array of complex numbers
+------------------------------------------
+
+That's actually big news, Pythran now decently support operations on array of  ``complex64``, ``complex128`` and ``complex256`` (if the backend compiler supports ``long double``) types.
+
+Verbose mode in ``pythran-config`` to debug compiler backend issues
+-------------------------------------------------------------------
+
+In some cases, knowing about the exact configuration files being loaded by Pythran helps debugging the setup. After all there's the default config file, the one living in your home, or maybe in ``XDG_CONFIG_HOME`` and the one specified by ``PYTHRANRC``. If in doublt, just run
+
+.. code-block:: shell
+
+    pythran-config -v
+
+And everything should be crystal-clear.
+
+Config file linting
+-------------------
+
+With that feature, any typo in the config file should now appear, well as a typo and not being silently ignored.
+
+Evaluate numpy.arange lazily when valid
+---------------------------------------
+
+Another optimization some people may appreciate: The pythran compiler can decide to evaluate ``np.arange`` lazily to avoid the array allocation, as in
+
+.. code-block:: python
+
+    def even_numbers(n):
+        return np.arange(n) * 2
+
+In that case Pythran only creates the end array, not the temporary one.
+
+Faster PRNG, namely pcg
+-----------------------
+
+I know that random numbers are a sloppy ground. Random numbers in Pythran have never strictly respected the semantic of Numpy's PRNG, that is we never produced the same sequence for the same seed. The previous engine was ``std::mt19937`` from the STL, it's now `PCG <http://www.pcg-random.org/>`_, and there's no guarantee it won't change in the future.
+
+Favor Python3 support in various places
+---------------------------------------
+
+Rember the Python3 statement from the beginning of this post?
+
+Fix Numpy.remainder implementation
+----------------------------------
+
+That was a funny one: ``std::remainder`` from C++ and ``numpy.remainder`` don't have the same semantic when dealing with negative numbers.
```

Comment:
"Don't yield identical results with negative numbers" (I don't think it's a question of semantics). You should confirm that pythran now does what numpy does, not what std::remainder does...